File: services/user-management/src/controllers/dev.controller.ts
import express, { Express } from "express"
import { connect } from "@/repositories/datasource"

const controller = express()

controller.post("/test-connection", async (_req, res) => {
  try {
    await connect().raw('select 1+1 as result')
    res.status(200).json({ message: "Successfully connected to database" })
  } catch {
    res.status(500).json({ message: "Failed to connect to database" })
  }
})

export const devController = {
  attach: (attacher: (controller: Express, path: string) => void) => attacher(controller, "dev")
}

File: services/event-scheduler/EventScheduler.Api/Controllers/DevController.cs
using EventScheduler.Db.Interfaces;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace EventScheduler.Api.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class DevController(IAppDbContext dbContext) : Controller
    {
        [HttpPost("/test-connection", Name = "TestConnection")]
        public IActionResult TestConnection()
        {
            return dbContext.Database.CanConnect()
                ? Ok(new { Message = "Successfully connected to database" })
                : StatusCode(500, new { Message = "Failed to connect to database" });
        }

        [HttpPost("/migrate", Name = "Migrate")]
        public IActionResult Migrate()
        {
            try
            {
                dbContext.Database.Migrate();
                return Ok(new { Message = "Successfully ran migrations" });
            }
            catch
            {
                return StatusCode(500, new { Message = "Failed to run migrations" });
            }

        }
    }
}
